### PR TITLE
Add Authentication and Comment Backend

### PR DESCRIPTION
**Pull request by JoshLew7**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public Comment(String name, String email, String description, long timestamp) {
+        this.name = name;
+        this.email = email;
+        this.description = description;
+        this.timestamp = timestamp;
+    }
+
+    public String toString(){
+        return "\n" + description;
+    }
```
Comment (chaowangzj) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:34]: Is this method used anywhere? Why does it only contain description?

File: portfolio/src/main/webapp/style.css [hunk 2]
```diff
@@ -382,6 +382,39 @@ ul li ul li{
...
+    margin-bottom: 2rem;
+    text-align: center;
+    color: #333333;
+}
+
+#special-features h2 .text-primary{
+    color: goldenrod;
+}
+
+#special-features .button{
```
Comment (xiangchengz) [inline portfolio/src/main/webapp/style.css:412]: keep the style attributes sorted.

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (xiangchengz) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:3]: curious, why we have this comment for Comment class but not for others in the same package?

File: portfolio/pom.xml [hunk 4]
```diff
@@ -35,7 +59,7 @@
         <version>2.2.0</version>
         <configuration>
           <!-- TODO: set project ID. -->
```
Comment (xiangchengz) [inline portfolio/pom.xml]: remove this to-do

File: portfolio/pom.xml [hunk 5]
```diff
@@ -23,8 +23,32 @@
...
+      <version>1.9.59</version>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.code.gson</groupId>
+      <artifactId>gson</artifactId>
+      <version>2.8.6</version>
+    </dependency>
+
+    <dependency>
```
Comment (xiangchengz) [inline portfolio/pom.xml]: why duplicate the maven dependency?

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -65,3 +65,23 @@ function addQuoteToDom(quote) {
...
+          alert("Error: Enter a valid comment.");
+        }
+      }
+  );
+}
+
+async function deleteComments() {
+  await fetch("/delete-data", { method: "POST" });
+  displayServletContent(0);
+}
```
Comment (chaowangzj) [inline portfolio/src/main/webapp/script.js:87]: You don't have the endpoint for "/delete-data" yet, and it is not used anywhere, is it something you want to add later?

File: portfolio/pom.xml (hunk 5 above)
Comment (chaowangzj) [reply portfolio/pom.xml]: +1, you can remove the redundant dependency.

File: portfolio/src/main/webapp/style.css (hunk 2 above)
Comment (JoshLew7) [reply portfolio/src/main/webapp/style.css:412]: CSS

File: portfolio/pom.xml (hunk 5 above)
Comment (JoshLew7) [reply portfolio/pom.xml]: I'm not too sure I guess I just hit copy paste twice

Comment (JoshLew7) [reply portfolio/pom.xml]: Dependecies

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (JoshLew7) [reply portfolio/src/main/webapp/script.js:87]: Wasn't used yet but I added it in.

Comment (JoshLew7) [reply portfolio/src/main/webapp/script.js:87]: Del

**Review by chaowangzj — COMMENTED** [review]
Hi Josh, can you commit the change you just made into this pull request instead of taking screenshots? I believe these changes are still in your local repo.

**Comment by JoshLew7** [comment]
> Hi Josh, can you commit the change you just made into this pull request instead of taking screenshots? I believe these changes are still in your local repo.

Yeah sorry I was trying to make it to where you didn't have to look far for the changes

**Comment by xiangchengz** [comment]
hi Josh, could you take time to resolve the left comments?

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 3 above)
Comment (JoshLew7) [reply portfolio/src/main/java/com/google/sps/data/Comment.java:3]: This was the class that was provided in the resource so I am not too sure.

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 1 above)
Comment (JoshLew7) [reply portfolio/src/main/java/com/google/sps/data/Comment.java:34]: This is used to display the comment content.

**Review by xiangchengz — APPROVED** [review]

**Pull request merged**